src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -600,10 +601,10 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
                         // the misalignment that may occur after; to we default to giving the GC a free hand to relocate and
                         // its up to the caller whether they are operating over fixed data.
                         Vector128<ushort> values = Vector128.Create((ushort)value);
-                        Vector128<ushort> search = LoadVector128(ref searchSpace, offset);
+                        Vector128<ushort> search = Vector128.LoadUnsafe(ref ushortSearchSpace, (nuint)offset);
 
                         // Same method as below
-                        int matches = Sse2.MoveMask(Sse2.CompareEqual(values, search).AsByte());
+                        uint matches = Vector128.Equals(values, search).AsByte().ExtractMostSignificantBits();
```

Comment:
Would it make sense to compare against `Vector128<byte>.Zero` and use `ExtractMostSignificantBits` only in the `else` branch? (asking based on the ARM64 benchmarks in https://github.com/dotnet/runtime/pull/73364).

(same for the other match below)

Reply:
Ah, nvm, I see this path needs Vector256 so it doesn't get hit on ARM and the Vector128 path already does that.